### ISSUE TITLE
Windows does not support MemorySwap

### ISSUE THREAD

**Issue by liangjl76** [hashicorp/nomad#2193]
### Nomad version
0.5.2

### Operating system and Environment details
Windows2016
Microsoft Windows [版本 10.0.14393]
docker  1.13.0-rc6

### Issue
01/13/17 08:27:07 CST  Driver Failure  failed to start task 'memcached' for alloc '446a721d-2575-cedf-2634-d2beaf9804e8': Failed to create container: API error (500): {"message":"invalid option: Windows does not support MemorySwap"}

**Comment by solarnz** [comment]
Is there any workaround available for this issue?

I didn't have this issue on docker 1.12, but after I upgraded to 1.13, I am now encountering this issue.

**Comment by dadgar** [comment]
We will get this fixed for 0.5.5

**Comment by cvandal** [comment]
+1

**Comment by mgarrettm** [comment]
+1

**Comment by schmichael** [comment]
Fixed by #2342 but on my test Windows 2016 server with Docker 1.13.1-cs1 installed I'm hitting a panic when trying to start the `redis:3.0-windowsservercore` container:

```
fatal error: unexpected signal during runtime execution
[signal 0xc0000005 code=0x0 addr=0xffffffffffffffff pc=0x40e892]

goroutine 185 [running]:
runtime.throw(0x1283a40, 0x2a)
        /usr/local/go/src/runtime/panic.go:596 +0x9c fp=0xc04213fe18 sp=0xc04213fdf8
runtime.sigpanic()
        /usr/local/go/src/runtime/signal_windows.go:155 +0x18b fp=0xc04213fe48 sp=0xc04213fe18
runtime.unlock(0xc042303092)
        /usr/local/go/src/runtime/lock_sema.go:107 +0x72 fp=0xc04213fe70 sp=0xc04213fe48
runtime.chansend(0x1039460, 0xc04230303a, 0xc04213ffb0, 0xc04213ff01, 0xabc44a, 0xc04213ff9c)
        /usr/local/go/src/runtime/chan.go:179 +0x8af fp=0xc04213ff20 sp=0xc04213fe70
runtime.chansend1(0x1039460, 0xc04230303a, 0xc04213ffb0)
        /usr/local/go/src/runtime/chan.go:113 +0x4d fp=0xc04213ff60 sp=0xc04213ff20
github.com/hashicorp/nomad/vendor/github.com/Microsoft/go-winio.ioCompletionProcessor(0x384)
        /home/schmichael/go/src/github.com/hashicorp/nomad/vendor/github.com/Microsoft/go-winio/file.go:131 +0xea fp=0xc04213ffd8 sp=0xc04213ff60
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:2197 +0x1 fp=0xc04213ffe0 sp=0xc04213ffd8
created by github.com/hashicorp/nomad/vendor/github.com/Microsoft/go-winio.initIo
        /home/schmichael/go/src/github.com/hashicorp/nomad/vendor/github.com/Microsoft/go-winio/file.go:55 +0x87
```

docker/docker#27423 sounds similar, but I don't have any non-IFS Winsock LSPs installed. This is a fresh Windows 2016 Server!

I built an amd64 Windows binary with the fix and attached it. I'd appreciate it if another Windows user could test it out and see if (a) it fixes this bug and (b) you hit the panic I'm hitting.

[nomad.exe.zip](https://github.com/hashicorp/nomad/files/791532/nomad.exe.zip)

**Comment by schmichael** [comment]
Added a new issue for the crasher. I tried updating the relevant packages and rebuilding but ran into the same crasher. Would love for a more experienced Windows user to test as well. See #2343
